File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> isomorphicMap = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char sChar = s.charAt(i);
            char tChar = t.charAt(i);

            if (isomorphicMap.containsKey(sChar)) {
                if (isomorphicMap.get(sChar) != tChar) {
                    return false;
                }
            } else if (isomorphicMap.containsValue(tChar)) {
                return false;
            } else {
                isomorphicMap.put(sChar, tChar);
            }
        }

        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.kt
class Solution {
    fun isIsomorphic(s: String, t: String): Boolean {
        val ismorphicMap = mutableMapOf<Char, Char>()

        for (i in s.indices) {
            val sChar = s[i]
            val tChar = t[i]

            when {
                ismorphicMap.containsKey(sChar) -> {
                    if (ismorphicMap[sChar] != tChar) return false
                }
                ismorphicMap.containsValue(tChar) -> return false
                else -> ismorphicMap[sChar] = tChar
            }
        }

        return true
    }
}